### PR TITLE
Issue.488 - Updating Kitura Docs

### PR DESCRIPTION
**Pull request by Andrew-Lees11**
## Description
Updated the Jazzy Docs for Kitura including a usage example for each class. This involves no changes to the code since it is only changing comments. The new Docs require review to ensure they properly describe the classes as well as general wording and grammar checks.

### CODE REVIEW

**Comment by helenmasters** [comment]
Andy, please can you fix the fact that the contributing guidelines link in the README.md file doesn't work from the context of the API reference site (it does work when used directly from the README.md file on github). I realise that this isn't one of the files you've changed but to my mind it forms part of this documentation work so should be fixed as part of this changeset.

File: Sources/Kitura/bodyParser/BodyParser.swift [hunk 1]
```diff
@@ -21,11 +21,13 @@ import Foundation
 
 // MARK: BodyParser
 
-/// The BodyParser parses the body of the request prior to sending it to the handler. It reads the Content-Type of the message header and populates the `RouterRequest` body field with a `ParsedBody` enumeration (e.g. json, raw, text, urlEncoded)
-/// In order for the BodyParser to be used it must first be registered with any routes that are interested in the ParsedBody payload. The example below registers all routes to the BodyParser middleware.
+/// The `BodyParser` parses the body of the request prior to sending it to the handler. It reads the Content-Type of the message header and populates the `RouterRequest` body field with a `ParsedBody` enumeration (e.g. json, raw, text, urlEncoded)
+/// In order for the BodyParser to be used it must first be registered with any routes that are interested in the ParsedBody payload.
```
Comment (helenmasters) [inline Sources/Kitura/bodyParser/BodyParser.swift]: Missing a full stop at the end of line 24.

File: Sources/Kitura/bodyParser/BodyParser.swift [hunk 2]
```diff
@@ -115,9 +117,9 @@ public class BodyParser: RouterMiddleware {
 
     /// Read incoming message for Parse
     ///
-    /// - Parameter message: message coming from the socket
+    /// - Parameter message: Message coming from the socket
     /// - Parameter parser: ((NSData) -> ParsedBody?) store at parserMap
-    /// - Returns: the parsed body
+    /// - Returns: The parsed body
```
Comment (helenmasters) [inline Sources/Kitura/bodyParser/BodyParser.swift]: Why are there no usage examples for the init, parse, readBodyData and handle calls?

File: Sources/Kitura/bodyParser/BodyParser.swift [hunk 3]
```diff
@@ -131,9 +133,9 @@ public class BodyParser: RouterMiddleware {
 
     /// Read the body data of the request.
     ///
-    /// - Parameter with: the socket reader.
+    /// - Parameter with: The socket reader.
     /// - Throws: Socket.Error if an error occurred while reading from a socket.
-    /// - Returns: data for the body.
+    /// - Returns: Data for the body.
```
Comment (helenmasters) [inline Sources/Kitura/bodyParser/BodyParser.swift]: "Data for the body." doesn't make sense to me how about something like "The body data of the request."?

File: Sources/Kitura/contentType/ContentType.swift [hunk 4]
```diff
@@ -19,30 +19,37 @@ import LoggerAPI
 
 // MARK: ContentType
 
-/// A set of APIs to work with Content-Type headers, whether to generate the value
-/// or to determine if it's an acceptable value.
+/**
+The `ContentType` class provides functions to determine the MIME content type for a given file extension. The user can pass in a complete file name e.g. "foo.png" or just the file extension e.g. "png", or they can pass in both a MIME content type and a file extension and query whether the they match.
```
Comment (helenmasters) [inline Sources/Kitura/contentType/ContentType.swift]: Remove "the" from "the they match"

File: Sources/Kitura/Kitura.swift [hunk 5]
```diff
@@ -22,7 +22,20 @@ import Dispatch
...
+```swift
+let router = Router()
+let port: Int = 8080
+public func run() throws{
+    ...
+    Kitura.addHTTPServer(onPort: port, with: router)
+    Kitura.run()
+}
+```
+In this example, we create a run function inside a server Application.swift file, which will create a Kitura server on the specified port, using the given router and then run the created server.
```
Comment (helenmasters) [inline Sources/Kitura/Kitura.swift]: The example text needs to go above the code snippet.

Comment (helenmasters) [reply Sources/Kitura/Kitura.swift]: @tunniclm or @rolivieri -  I notice that we are saying two different things in this file - in the example code we call Kitura run() function inside a server Application.swift file, however, within the run() function we have a note: This function never returns - it should be the last call in your main.swift file.
Where is the recommended place to put the Kitura run() call as I'd like our usage example to reflect this (or definitely to have some better wording as to why either location may be suitable)?

Comment (helenmasters) [reply Sources/Kitura/Kitura.swift]: The following sentences need full stops at the end - "Stop all registered servers", "Start all registered servers and return"

Comment (helenmasters) [reply Sources/Kitura/Kitura.swift]: /// - Parameter unregister: If servers should be unregistered after stopped (default true).

Please add the words - they are - so it reads unregistered after they are stopped.

**Comment by ianpartridge** [comment]
@Andrew-Lees11 this is stale - what's the status?

**Comment by djones6** [comment]
Closing after discussion: this is being split into smaller updates.

**Comment by ianpartridge** [comment]
Do we need to keep the branch around?

**Pull request closed without merging**